pkg/detector/detector.go

```diff
@@ -231,7 +235,7 @@ func (d *ResourceDetector) Reconcile(key util.QueueKey) error {
 			klog.Infof("Waiting for dependent overrides present for policy(%s/%s)", propagationPolicy.Namespace, propagationPolicy.Name)
 			return fmt.Errorf("waiting for dependent overrides")
 		}
-
+		d.RemoveWaiting(clusterWideKey)
```

Comment:
What if `AppliPolicy` failed?

Reply:
Then it would requeue and reconcile again, right?

Reply:
get it.